File: json/json-annotation-processor/src/main/java/io/koraframework/json/annotation/processor/JsonTypes.java
package io.koraframework.json.annotation.processor;

import com.palantir.javapoet.ClassName;

public class JsonTypes {
    public static final ClassName json = ClassName.get("io.koraframework.json.common.annotation", "Json");
    public static final ClassName jsonInclude = ClassName.get("io.koraframework.json.common.annotation", "JsonInclude");
    public static final ClassName jsonDiscriminatorField = ClassName.get("io.koraframework.json.common.annotation", "JsonDiscriminatorField");
    public static final ClassName jsonDiscriminatorValue = ClassName.get("io.koraframework.json.common.annotation", "JsonDiscriminatorValue");

    public static final ClassName jsonReaderAnnotation = ClassName.get("io.koraframework.json.common.annotation", "JsonReader");
    public static final ClassName jsonWriterAnnotation = ClassName.get("io.koraframework.json.common.annotation", "JsonWriter");

    public static final ClassName jsonNullable = ClassName.get("io.koraframework.json.common", "JsonNullable");
    public static final ClassName jsonReader = ClassName.get("io.koraframework.json.common", "JsonReader");
    public static final ClassName jsonWriter = ClassName.get("io.koraframework.json.common", "JsonWriter");

    public static final ClassName enumJsonReader = ClassName.get("io.koraframework.json.common", "EnumJsonReader");
    public static final ClassName enumJsonWriter = ClassName.get("io.koraframework.json.common", "EnumJsonWriter");

    public static final ClassName jsonFieldAnnotation = ClassName.get("io.koraframework.json.common.annotation", "JsonField");
    public static final ClassName jsonSkipAnnotation = ClassName.get("io.koraframework.json.common.annotation", "JsonSkip");

    public static final ClassName bufferingJsonParser = ClassName.get("io.koraframework.json.common.util", "BufferingJsonParser");
    public static final ClassName discriminatorHelper = ClassName.get("io.koraframework.json.common.util", "DiscriminatorHelper");

    public static final ClassName jsonParseException = ClassName.get("tools.jackson.core.exc", "StreamReadException");
    public static final ClassName jsonParser = ClassName.get("tools.jackson.core", "JsonParser");
    public static final ClassName jsonParserSequence = ClassName.get("tools.jackson.core.util", "JsonParserSequence");
    public static final ClassName jsonGenerator = ClassName.get("tools.jackson.core", "JsonGenerator");
    public static final ClassName jsonToken = ClassName.get("tools.jackson.core", "JsonToken");
    public static final ClassName serializedString = ClassName.get("tools.jackson.core.io", "SerializedString");

}

File: json/json-symbol-processor/src/main/kotlin/io/koraframework/json/ksp/JsonTypes.kt
package io.koraframework.json.ksp

import com.squareup.kotlinpoet.ClassName

object JsonTypes {

    val json = ClassName("io.koraframework.json.common.annotation", "Json")
    val jsonInclude = ClassName("io.koraframework.json.common.annotation", "JsonInclude")
    val jsonDiscriminatorField = ClassName("io.koraframework.json.common.annotation", "JsonDiscriminatorField")
    val jsonDiscriminatorValue = ClassName("io.koraframework.json.common.annotation", "JsonDiscriminatorValue")

    val jsonNullable = ClassName("io.koraframework.json.common", "JsonNullable")
    val jsonReaderAnnotation = ClassName("io.koraframework.json.common.annotation", "JsonReader")
    val jsonWriterAnnotation = ClassName("io.koraframework.json.common.annotation", "JsonWriter")

    val jsonReader = ClassName("io.koraframework.json.common", "JsonReader")
    val jsonWriter = ClassName("io.koraframework.json.common", "JsonWriter")

    val enumJsonReader = ClassName("io.koraframework.json.common", "EnumJsonReader")
    val enumJsonWriter = ClassName("io.koraframework.json.common", "EnumJsonWriter")

    val jsonFieldAnnotation = ClassName("io.koraframework.json.common.annotation", "JsonField")
    val jsonSkipAnnotation = ClassName("io.koraframework.json.common.annotation", "JsonSkip")

    val bufferingJsonParser = ClassName("io.koraframework.json.common.util", "BufferingJsonParser");
    val discriminatorHelper = ClassName("io.koraframework.json.common.util", "DiscriminatorHelper");

    val jsonParseException = ClassName("tools.jackson.core.exc", "StreamReadException")
    val jsonParser = ClassName("tools.jackson.core", "JsonParser")
    val jsonGenerator = ClassName("tools.jackson.core", "JsonGenerator")
    val jsonToken = ClassName("tools.jackson.core", "JsonToken")
    val jsonParserSequence = ClassName("tools.jackson.core.util", "JsonParserSequence")
    val serializedString = ClassName("tools.jackson.core.io", "SerializedString")
}
